src/api/registry.js

```diff
@@ -160,22 +160,19 @@ export async function setContent(name, content) {
   const namehash = await getNamehash(name)
   const resolverAddr = await getResolver(name)
   const { Resolver } = await getResolverContract(resolverAddr)
-  // const gas = await Resolver.setName(namehash, content).estimateGas()
-  // console.log('gas', gas)
-  return () => Resolver.setContent(namehash, content).send({ from: account })
-  // return () => Resolver.setContent(namehash, content).send({ from: account, gas:gas })
+  const gas = await Resolver.setContent(namehash, content).estimateGas()
+  return () =>
+    Resolver.setContent(namehash, content).send({ from: account, gas })
 }
 
 export async function setContenthash(name, content) {
   const account = await getAccount()
   const namehash = await getNamehash(name)
   const resolverAddr = await getResolver(name)
   const { Resolver } = await getResolverContract(resolverAddr)
-  // const gas = await Resolver.setContenthash(namehash, content).estimateGas()
-  // console.log('gas', gas)
-  // return () => Resolver.setContenthash(namehash, content).send({ from: account, gas:gas })
-  return () =>
-    Resolver.setContenthash(namehash, content).send({ from: account })
+  const tx = Resolver.setContenthash(namehash, content)
+  const gas = await tx.estimateGas()
```

Comment:
@jefflau I remember I mentioned this a few weeks ago.
If you use estimate gas, it works on ganache but for some reason it fails on testnet when setting contenthash (probably same for content as well)
<img width="1494" alt="Screenshot 2019-04-01 at 18 23 15" src="https://user-images.githubusercontent.com/2630/55346832-5264f680-54ab-11e9-8484-20d004bda52a.png">

Reply:
Damn okay, I guess we have to remove it then